llpc/context/llpcCompiler.cpp

```diff
@@ -728,6 +728,246 @@ template <typename T> static FormattedBytes formatBytesLittleEndian(ArrayRef<T>
   return format_bytes(bytes, /* FirstByteOffset = */ None, /* NumPerLine = */ 16, /* ByteGroupSize = */ 1);
 }
 
+// Build shader from the specified info.
+//
+// @param [in]  pipelineInfo : Info to build this shader module
+// @param [out] pipelineOut : Output of building this shader module
+// @param [in]  stage : Shader stage of needing to compile
+// @param pPipelineDumpFile : Handle of pipeline dump file
+//
+// @returns : Result::Success if successful. Other return codes indicate failure.
+Result Compiler::buildGraphicsShaderStage(const GraphicsPipelineBuildInfo *pipelineInfo,
+                                          GraphicsPipelineBuildOut *pipelineOut, UnlinkedShaderStage stage,
+                                          void *pPipelineDumpFile) {
+  Result result = Result::Success;
```

Comment:
Don't define `result` until it's necessary. Prefer early returns when possible. See https://github.com/GPUOpen-Drivers/llpc/blob/dev/docs/CodingStandards.md#result-handling